rust/types/src/spann_posting_list.rs

```diff
@@ -0,0 +1,17 @@
+pub struct SpannPostingList<'referred_data> {
+    pub doc_offset_ids: &'referred_data [u32],
+    pub doc_versions: &'referred_data [u32],
+    // Flattened out embeddings for all documents.
```

Comment:
Assuming that we assume dimension is bookkeeped elsewhere?

Reply:
Well the dimension is assumed to be `doc_embeddings.len() / doc_offset_ids.len()` lol. Curious, do you think of that as brittle and error prone?